diff -ur ../old/MG5_aMC_v2_2_3/madgraph/various/cluster.py MG5_aMC_v2_2_3/madgraph/various/cluster.py
--- ../old/MG5_aMC_v2_2_3/madgraph/various/cluster.py	2015-02-10 03:55:06.000000001 +0100
+++ MG5_aMC_v2_2_3/madgraph/various/cluster.py	2015-02-24 12:41:08.000000001 +0100
@@ -346,14 +346,14 @@
             with option: %s
             file missing: %s.
             Stopping all runs.''' % (job_id, args, path))
-            #self.remove()
+            raise ClusterManagmentError('Job exhausted all retries.')
         elif args['nb_submit'] >= self.nb_retry:
             logger.critical('''Fail to run correctly job %s.
             with option: %s
             file missing: %s
             Fails %s times
             No resubmition. ''' % (job_id, args, path, args['nb_submit']))
-            #self.remove()
+            raise ClusterManagmentError('Job exhausted all retries.')
         else:
             args['nb_submit'] += 1            
             logger.warning('resubmit job (for the %s times)' % args['nb_submit'])
@@ -1323,6 +1323,8 @@
         if log is None:
             log = '/dev/null'
         
+        text += 'if [ -n $CMSSW_BASE ]; then cd $CMSSW_BASE; eval `scramv1 runtime -sh`; cd -; fi;'
+        
         text += prog
         if argument:
             text += ' ' + ' '.join(argument)
diff -ur ../old/MG5_aMC_v2_2_3/Template/LO/bin/internal/Gridpack/run.sh MG5_aMC_v2_2_3/Template/LO/bin/internal/Gridpack/run.sh
--- ../old/MG5_aMC_v2_2_3/Template/LO/bin/internal/Gridpack/run.sh	2015-02-10 03:55:06.000000001 +0100
+++ MG5_aMC_v2_2_3/Template/LO/bin/internal/Gridpack/run.sh	2015-02-24 12:38:36.000000001 +0100
@@ -78,37 +78,5 @@
     cd ..
 fi
 
-if [[ -e ./DECAY/decay ]]; then
-    cd DECAY
-    echo -$seed > iseed.dat
-    for ((i = 1 ;  i <= 20;  i++)) ; do
-	if [[ -e decay_$i\.in ]]; then
-	    echo "Decaying events..."
-	    mv ../events.lhe ../events_in.lhe
-	    ./decay < decay_$i\.in
-	fi
-    done
-    cd ..
-fi
-
-if [[ -e ./REPLACE/replace.pl ]]; then
-    for ((i = 1 ;  i <= 20;  i++)) ; do
-	if [[ -e ./REPLACE/replace_card$i\.dat ]];then
-	    echo "Adding flavors..."
-	    mv ./events.lhe ./events_in.lhe
-	    cd ./REPLACE
-	    ./replace.pl ../events_in.lhe ../events.lhe < replace_card$i\.dat
-	    cd ..
-	fi
-    done
-fi
-
-# part added by Stephen Mrenna to correct the kinematics of the replaced
-#  particles
-if [[ -e ./madevent/bin/internal/addmasses.py ]]; then
-  mv ./events.lhe ./events.lhe.0
-  python ./madevent/bin/internal/addmasses.py ./events.lhe.0 ./events.lhe
-fi  
-
 gzip -f events.lhe
 exit
